runtime/bundles/org.eclipse.core.jobs/src/org/eclipse/core/runtime/jobs/Job.java

```diff
@@ -194,6 +196,82 @@ protected IStatus run(IProgressMonitor monitor) {
 		};
 	}
 
+	/**
+	 * Performs a one shot background job execution returning a value as an
+	 * syncronous computation to the caller with the follwoing side-effects:
+	 * <ul>
+	 * <li>If the {@link CompletableFuture} is canceled, best effort is made to also
+	 * cancel the job</li>
+	 * <li>If the backing {@link Job} is canceled, best effort it made to also
+	 * cancel the {@link CompletableFuture}</li>
+	 * </ul>
+	 *
+	 * @since 3.14
+	 */
+	public static CompletableFuture<?> execute(String name, IJobRunnable runnable) {
+		return compute(name, monitor -> {
+			runnable.run(monitor);
+			return null;
+		});
+	}
+
+	/**
+	 * Performs a one shot background job execution returning a value as an
+	 * syncronous computation to the caller with the follwoing side-effects:
+	 * <ul>
+	 * <li>If the {@link CompletableFuture} is canceled, best effort is made to also
+	 * cancel the job</li>
+	 * <li>If the backing {@link Job} is canceled, best effort it made to also
+	 * cancel the {@link CompletableFuture}</li>
+	 * </ul>
+	 *
+	 * @since 3.14
+	 */
+	public static <T> CompletableFuture<T> compute(String name, IJobCallable<T> callable) {
+		CompletableFuture<T> future = new CompletableFuture<>();
```

Comment:
I don't think a cancellation request on the future is propagated to the job yet, is it? Please add a test that illustrates this.

Reply:
A, my bad: I oversaw the future.exceptionally(..) below.